src/lambda/commands/uploadLambda.ts

```diff
@@ -43,24 +45,25 @@ interface LambdaFunction {
  * Executes the "Upload Lambda..." command.
  * Allows for uploads of zip files, and both built and unbuilt directories.
  * Does not discriminate on runtime.
- * @param functionNode Function node from AWS Explorer
+ * @param lambdaArg LambdaFunction
+ * @param path Uri to the template.yaml file or the directory the command was invoked from
  */
-export async function uploadLambdaCommand(functionNode: LambdaFunctionNode) {
+export async function uploadLambdaCommand(lambdaArg?: LambdaFunction, path?: vscode.Uri) {
     let result: telemetry.Result = 'Cancelled'
-    const lambda = {
-        name: functionNode.functionName,
-        region: functionNode.regionCode,
-        configuration: functionNode.configuration,
-    }
+    let lambda: LambdaFunction | undefined
 
     try {
-        const response = await new UploadLambdaWizard(lambda).run()
-
-        if (response?.uploadType === 'zip') {
+        const response = await new UploadLambdaWizard(lambdaArg, path).run()
+        if (!response) {
+            getLogger().debug('lambda: UploadLambdaWizard returned undefined. User cancelled.')
+            return
+        }
+        lambda = response.lambda
+        if (response.uploadType === 'zip') {
             await runUploadLambdaZipFile(lambda, response.targetUri)
             result = 'Succeeded'
-        } else if (response?.uploadType === 'directory' && response.directoryUploadType) {
-            result = (await runUploadDirectory(lambda, response.directoryUploadType, response.targetUri)) ?? result
+        } else if (response.uploadType === 'directory' && response.directoryBuildType) {
+            result = (await runUploadDirectory(lambda, response.directoryBuildType, response.targetUri)) ?? result
             result = 'Succeeded'
```

Comment:
Looks like a mistake in the old code: `result` is assigned in the line above this, then immediately overwritten with 'Succeeded'. @JadenSimon

Reply:
Going to leave this as-is for now because the result needs to be "Succeeded".

Reply:
Was a merge artifact. The legacy `runUploadLambdaWithSamBuild` can potentially show a prompt which may cancel the operation. It should be `?? 'Succeeded'` not `?? result`.

Minor thing though. You'd just lose a bit of nuance in the telemetry.